sites/all/modules/custom/training2_form/training2_form.module

```diff
@@ -0,0 +1,182 @@
+<?php
+
+/**
+ * @file
+ * Homework on topic Block and Menu.
+ */
+
+/**
+ * Implements hook_block_info().
+ */
+function training2_form_block_info() {
+  $blocks['currency'] = array(
+    'info' => t('Currency block'),
+    'cache' => DRUPAL_CACHE_GLOBAL,
+  );
+
+  return $blocks;
+}
+
+/**
+ * Implements hook_block_configure().
+ */
+function training2_form_block_configure($delta = '') {
+  $form = array();
+
+  if ($delta = 'currency') {
+    $arrCurrency = training2_form_conf_content();
+
+    $form['form_currency'] = array(
+      '#type' => 'checkboxes',
+      '#title' => t('Choose currencies'),
+      '#options' => drupal_map_assoc($arrCurrency),
+    );
+  }
+
+  return $form;
+}
+
+/**
+ * Implements hook_block_save().
+ */
+function training2_form_block_save($delta = '', $edit = array()) {
+
+  if ($delta == 'currency') {
+    variable_set('form_currency', $edit['form_currency']);
```

Comment:
Check $edit['form_currency'] for empty value. In the case it can cause php notice